Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/Home.kt

```diff
@@ -108,21 +109,57 @@ enum class HomeSections(
     PROFILE(R.string.home_profile, Icons.Outlined.AccountCircle, "home/profile")
 }
 
+/**
+ * Remembers and creates an instance of [BottomBarStateHolder]
+ */
+@Composable
+fun rememberBottomBarStateHolder(navController: NavController, tabs: Array<HomeSections>) =
+    remember(navController, tabs) {
+        BottomBarStateHolder(navController, tabs)
+    }
+
+/**
+ * Responsible for holding state related to [JetsnackBottomBar] and containing UI-related logic.
+ */
+@Stable
+class BottomBarStateHolder(
+    private val navController: NavController,
+    val tabs: Array<HomeSections>
+) {
+    private val sections = HomeSections.values()
+    private val currentRoute: String?
+        get() = navController.currentDestination?.route
+
+    val routes = sections.map { it.route }
+
+    // Reading this attribute will cause recompositions when the route becomes valid or invalid
+    val isValidRoute: Boolean
```

Comment:
What makes a route "valid"? Maybe add a comment to explain that the bottom bar should only be shown on specific pages, and this boolean checks if the current destination is in that set? Or even consider renaming to "shouldBeDisplayed" or something like that?

Reply:
Done, thanks!